Mathlib/CategoryTheory/ChosenFiniteProducts/Cat.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Nicolas Rolland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolas Rolland
+-/
+import Mathlib.CategoryTheory.ChosenFiniteProducts
+/-!
+# Chosen finite products in `Cat`
+
+This file proves that the cartesian product of a pair of categories agrees with the
+product in `Cat`, and provides the associated `ChosenFiniteProducts` instance.
+-/
+
+universe v u
+
+namespace CategoryTheory
+
+namespace Cat
+
+open Limits
+
+/-- The chosen terminal object in `Cat`. -/
+abbrev chosenTerminal : Cat := Cat.of (ULift (ULiftHom (Discrete Unit)))
+
+example : chosenTerminal := ⟨⟨⟨⟩⟩⟩
+
+/-- The chosen terminal object in `Cat` is terminal. -/
+def chosenTerminalIsTerminal : IsTerminal chosenTerminal :=
+  IsTerminal.ofUniqueHom (fun _ ↦ (Functor.const _).obj ⟨⟨⟨⟩⟩⟩) fun _ _ ↦ rfl
+
+/-- The chosen product of categories `C × D` yields a product cone in `Cat`. -/
+def prodCone (C D : Cat.{v,u}) : BinaryFan C D :=
+  .mk (P := .of (C × D)) (Prod.fst _ _) (Prod.snd _ _)
+
+/-- The product cone in `Cat` is indeed a product. -/
+def isLimitProdCone (X Y : Cat) : IsLimit (prodCone X Y) := BinaryFan.isLimitMk
+  (fun S => S.fst.prod' S.snd) (fun _ => rfl) (fun _ => rfl) (fun _ _ h1 h2 =>
+    Functor.hext
+      (fun _ ↦ Prod.ext (by simp [← h1]) (by simp [← h2]))
+      (fun _ _ _ ↦ by dsimp; rw [← h1, ← h2]; rfl))
+
+instance : ChosenFiniteProducts Cat where
+  product (X Y : Cat) := { isLimit := isLimitProdCone X Y }
+  terminal  := { isLimit := chosenTerminalIsTerminal }
+
+example : MonoidalCategory Cat := by infer_instance
```

Comment:
I don't these `example`'s are well-motivated. If the goal is to make sure this doesn't disappear, then they should be tests. If the goal is to communicate to the reader that they can be solved by typeclass synthesis, this should be a comment.

Reply:
I think I suggested putting this as an `example` (the author had written it as an `instance` initially). My idea was communication to the reader, is there a rule to not use examples for this? (I'm certainly guilty of a few such `example`s in the library)

Reply:
There is no policy per se. I think the module doc string is a better place to explain how these instances allow Lean to synthesize a `MonoidalCategory` instance for `Cat`. At the least, there should be a comment before the `example`s to explain what these are supposed to convey and why they are here.

Reply:
I will add a comment, it is clearer.